# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/animation/AnimationEventDispatcher.h
# Commit: 63603267402e
# Full Hash: 63603267402ee93d6551ef60c35191d99aaee06b
# Author: Zach Hoffman <zach@zrhoffman.net>
# Date: 2023-12-18 17:17:57
# Regressor Bug: 1846516
# File Overlap Count: 3
# Description:
#   Bug 1846516 - [css-properties-values-api] Introduce AnimatedPropertyID/AnimatedPropertIDSet. r=firefox-animation-reviewers,firefox-style-system-reviewers,layout-reviewers,emilio,hiro
#   
#   This will make possible to animate custom properties on Gecko side. For now, the
#   animation code keeps only dealing with nsCSSPropertyID, so behavior is unchanged.
#   
# ==============================================================================

diff -r 753bc39f59ff -r 63603267402e dom/animation/AnimationEventDispatcher.h
--- a/dom/animation/AnimationEventDispatcher.h	Mon Dec 18 09:24:07 2023 +0000
+++ b/dom/animation/AnimationEventDispatcher.h	Mon Dec 18 09:24:08 2023 +0000
@@ -124,7 +124,7 @@
 
   struct CssTransitionData : public CssAnimationOrTransitionData {
     // For transition events only.
-    const nsCSSPropertyID mPropertyId = eCSSProperty_UNKNOWN;
+    const AnimatedPropertyID mProperty;
   };
 
   struct WebAnimationData {
@@ -150,7 +150,7 @@
   void MaybeAddMarker() const;
 
   // For CSS animation events
-  AnimationEventInfo(nsAtom* aAnimationName,
+  AnimationEventInfo(RefPtr<nsAtom> aAnimationName,
                      const NonOwningAnimationTarget& aTarget,
                      EventMessage aMessage, double aElapsedTime,
                      const TimeStamp& aScheduledEventTimeStamp,
@@ -160,14 +160,14 @@
         mData(CssAnimationData{
             {OwningAnimationTarget(aTarget.mElement, aTarget.mPseudoType),
              aMessage, aElapsedTime},
-            aAnimationName}) {
+            std::move(aAnimationName)}) {
     if (profiler_thread_is_being_profiled_for_markers()) {
       MaybeAddMarker();
     }
   }
 
   // For CSS transition events
-  AnimationEventInfo(nsCSSPropertyID aProperty,
+  AnimationEventInfo(const AnimatedPropertyID& aProperty,
                      const NonOwningAnimationTarget& aTarget,
                      EventMessage aMessage, double aElapsedTime,
                      const TimeStamp& aScheduledEventTimeStamp,
@@ -242,8 +242,7 @@
       }
 
       InternalTransitionEvent event(true, data.mMessage);
-      CopyUTF8toUTF16(nsCSSProps::GetStringValue(data.mPropertyId),
-                      event.mPropertyName);
+      data.mProperty.ToString(event.mPropertyName);
       event.mElapsedTime = data.mElapsedTime;
       event.mPseudoElement =
           nsCSSPseudoElements::PseudoTypeAsString(data.mTarget.mPseudoType);